### DISCUSSION TITLE
PxSceneDesc.isValid() throws unresolved external symbol

### DISCUSSION THREAD

**Discussion by Freznosis** [NVIDIA-Omniverse/PhysX#110] [Q&A] [answered]
I am trying to check if a **PxSceneDesc** is valid after setting it up with **PxSceneDesc.isValid()** but it throws an unresolved external symbol error. Seems to not like something about **PxgDynamicsMemoryConfig** when checking **gpuDynamicsConfig.isValid()**? Am I missing a certain GPU include?

`unresolved external symbol "__declspec(dllimport) public: bool __cdecl physx::PxgDynamicsMemoryConfig::isValid(void)const " (__imp_?isValid@PxgDynamicsMemoryConfig@physx@@QEBA_NXZ) referenced in function "public: class physx::PxScene * __cdecl PxlibWorld::CreatePhysicsScene(void)" (?CreatePhysicsScene@PxlibWorld@@QEAAPEAVPxScene@physx@@XZ)`

_PhysX 5.1.3 dynamically linked._

**Comment by preist-nvidia** [comment]
The scene desk and gpu mem validation is all in the same header, see here: https://github.com/NVIDIA-Omniverse/PhysX/blob/release/104.2/physx/include/PxSceneDesc.h#L410. Can you try including it?

**Reply by Freznosis** [reply]
Yeah I saw that. I tried a lot of different combinations of using/include but could not get it to compile. My next wonder is if PX_SUPPORT_GPU_PHYSX is not being properly defined in the source project? I checked over the source in PxPreprocessor and it seems to be defined properly when building. So I'm at a loss here. Anything else I can check?

Edit: Oh I see PxgDynamicsMemoryConfig is defined whether PX_SUPPORT_GPU_PHYSX is defined or not... well there goes all of my ideas. :)

Edit2: From some more testing. I can use PxgDynamicsMemoryConfig and manipulate it and use it in my project just fine. But as soon as I call isValid() is when the error occurs. Would there be any reason isValid() is not available in a checked build?

![image](https://user-images.githubusercontent.com/63488230/223225179-258a630e-6365-4f61-b565-f3812863811d.png)


Thanks

**Reply by preist-nvidia** [reply]
I'm sorry, I am not sure how I can help further - debugging linker issues is out of the scope of what we can help with.

**Comment by Freznosis** [comment, marked as answer]
For future reference or anyone with a similar problem, adding a definition for **PX_PHYSX_STATIC_LIB** fixed this for me.

**Reply by preist-nvidia** [reply]
Glad you found the issue, @Freznosis - and thanks for reporting back here!
